### PR TITLE
Heuristics tests are altered to panic with multiple results

### PR DESCRIPTION
**Pull request by Lazark0x**
Resolves #4246

### CODE REVIEW

File: runtime/vara/src/tests.rs [hunk 1]
```diff
@@ -20,15 +20,345 @@ use super::*;
...
+/// Memory pages access costs.
+struct PagesCosts {
+    pub load_page_data: CostOf<GearPagesAmount>,
+    pub upload_page_data: CostOf<GearPagesAmount>,
+    pub mem_grow: CostOf<GearPagesAmount>,
+    pub mem_grow_per_page: CostOf<GearPagesAmount>,
+    pub parachain_read_heuristic: CostOf<GearPagesAmount>,
+}
+
+impl<T: pallet_gear::Config> From<MemoryWeights<T>> for PagesCosts {
```
Comment (StackOverflowExcept1on) [inline runtime/vara/src/tests.rs:93]: btw we already have `gear_core::costs::LazyPagesCosts`. mb reuse it here?
 https://github.com/gear-tech/gear/blob/73929e309d40aceec8da761b8b2313eab7ea41a4/pallets/gear/src/schedule.rs#L1304

Comment (StackOverflowExcept1on) [reply runtime/vara/src/tests.rs:93]: we could add the missing fields (from `MemoryWeights<T>`) to `gear_core::costs::LazyPagesCosts` and reuse that for both gtest and these checks.

File: runtime/vara/src/tests.rs [hunk 2]
```diff
@@ -20,15 +20,345 @@ use super::*;
...
+            WeightExpectation::new(
+                weights.$inst_name.into(),
+                expected.$inst_name.into(),
+                INSTRUCTIONS_SPREAD,
+                stringify!($inst_name),
+            )
+        };
+    }
+
+    let expectations = vec![
```
Comment (breathx) [inline runtime/vara/src/tests.rs:121]: It'd be great also to have some check of missing variants

File: runtime/vara/src/tests.rs (hunk 1 above)
Comment (Lazark0x) [reply runtime/vara/src/tests.rs]: Done

File: runtime/vara/src/tests.rs (hunk 2 above)
Comment (Lazark0x) [reply runtime/vara/src/tests.rs:321]: Yeah, added checks for the number of fields to be checked

File: runtime/vara/src/tests.rs (hunk 1 above)
Comment (breathx) [reply runtime/vara/src/tests.rs]: That way LazyPagesCosts become not actually costs of lazy pages. wdyt @grishasobol ?

Comment (Lazark0x) [reply runtime/vara/src/tests.rs]: Do you think we need a more generic name for the structure? @breathx

File: runtime/vara/src/tests.rs [hunk 3]
```diff
@@ -18,17 +18,547 @@
...
 use pallet_gear::{InstructionWeights, MemoryWeights, SyscallWeights};
 use pallet_staking::WeightInfo as _;
-use runtime_common::weights::{
-    check_instructions_weights, check_lazy_pages_costs, check_pages_costs, check_syscall_weights,
-    PagesCosts,
-};
 use sp_runtime::AccountId32;
 
+#[cfg(feature = "dev")]
+use frame_support::traits::StorageInstance;
```
Comment (breathx) [inline runtime/vara/src/tests.rs:30]: pls move everything into module. runtime/vara/tests is general testing module. so all "utils" like `WeightExpectation' should be somewhere else. keep here tests only

**Review by breathx — APPROVED** [review]
lgtm, Im only aware of naming for lazy pages costs, waiting for additional opinions here

File: runtime/vara/src/tests.rs (hunk 3 above)
Comment (Lazark0x) [reply runtime/vara/src/tests.rs:32]: Done

**Review by breathx — APPROVED** [review]

File: core/src/costs.rs [hunk 4]
```diff
@@ -502,6 +502,21 @@ impl SyscallCosts {
     }
 }
 
+/// Memory pages costs.
+#[derive(Debug, Default, Clone, PartialEq, Eq)]
+pub struct PagesCosts {
```
Comment (StackOverflowExcept1on) [inline core/src/costs.rs:507]: also need to update `TYPE_LIST` (add `PagesCosts`, `IoCosts`) and regenerate `core/src/gas_metering/schedule.rs`
https://github.com/gear-tech/gear/blob/b9511183910d6b1ff6a018a5cadb72dc784b4fee/utils/weight-diff/src/main.rs#L305

Comment (Lazark0x) [reply core/src/costs.rs:507]: Done

**Review by StackOverflowExcept1on — APPROVED** [review]

**Pull request merged**